models/fates/R/write.configs.FATES.R

```diff
@@ -203,10 +242,404 @@ write.config.FATES <- function(defaults, trait.values, settings, run.id){
+       }
+       if(var == "fdi_a"){            ## SPITFIRE Constant in calculation of dewpoint for Fire Danger Index (FDI)
+         ncvar_put(nc=param.nc, varid='fdi_a',vals=pft[v])
+       }
+       if(var == "fdi_b"){            ## SPITFIRE Constant in calculation of dewpoint for Fire Danger Index (FDI)
+         ncvar_put(nc=param.nc, varid='fdi_b',vals=pft[v])
+       }
+       # fdi_alpha	SPITFIRE fire danger index (FDI) coefficient	
+       if(var == ""){            ## 
+         ncvar_put(nc=param.nc, varid='',vals=pft[v])
+       }
+       ##   -- CANOPY
+       if(var == "canopy_max_spread"){            ## Maximum allowable "dynamic ratio of dbh to canopy area" for cohorts in closed canopies.	-	[cm/m2]
+         ncvar_put(nc=param.nc, varid='maxspread',vals=pft[v])
+       }
+       # 	
+       if(var == "canopy_min_spread"){            ## Minimum allowable "dynamic ratio of dbh to canopy area" for cohorts in closed canopies.	-	[cm/m2]
+         ncvar_put(nc=param.nc, varid='minspread',vals=pft[v])
+       }
+
+       ## LITTERCLASS indexed (Size:6)
+       ## MCD: skipping for now until there's demonstrated demand because it requires expanding every variable out into VARNAME_[1..n]
+       # low_moisture_C	Intercept (constant) of fuel moisture to burned fraction term for drier fuel	litterclass	
+       # low_moisture_S	Slope of fuel moisture to burned fraction term for drier fuel	litterclass	
+       # max_decomp	Maximum decomposition rate of litter in the absence of moisture or temperature stress, per fuel class	litterclass	y-1
+       # mid_moisture	Parameter of burned fraction term. Below this 'low' constants apply, above this, 'mid' constants apply, 	litterclass	
+       # mid_moisture_C	Intercept (constant) of fuel moisture to burned fraction term for wetter fuel	litterclass	
+       # min_moisture	Parameter of burned fraction term. Below this value all litter is burned by a fire. Above, 'low' constants apply	litterclass	
+       # FBD	Fuel Bulk Density of fuel class 	litterclass	kg m-3
+       # alpha_FMC	Parameter of function relating fuel moisture content to meteorological fire danger index 	litterclass	
+       # SAV	Surface Area to Volume Ratio of fuel class 	litterclass	cm-1
+       
+       ## NCWD dimensioned       Size:4
+       if(var == "CWD_frac1"){            ##Fraction of coarse woody debris (CWD) that is moved into each of the four woody fuel classes
+         ncvar_put(nc=param.nc, varid='CWD_frac', start = 1, count = 1,
+                   vals=pft[v])
+       }
+       if(var == "CWD_frac2"){
+         ncvar_put(nc=param.nc, varid='CWD_frac', start = 2, count = 1,
+                   vals=pft[v])
+       }
+       if(var == "CWD_frac3"){
+         ncvar_put(nc=param.nc, varid='CWD_frac', start = 3, count = 1,
+                   vals=pft[v])
+       }
+       if(var == "CWD_frac4"){
+         ncvar_put(nc=param.nc, varid='CWD_frac', start = 4, count = 1,
+                   vals=pft[v])
+       }
+       
```

Comment:
As a temporary hack, I guess this is OK, but this violates the hell out of the "Don't Repeat Yourself" (DRY) principle and is super exposed to bugs. It would be much better to define a `list` or `data.frame` of trait-specific arguments and then have a single function that does `do.call` on `ncvar_put`.